tests/test.bats

```diff
@@ -0,0 +1,340 @@
+  docker_sqlplus_admin "DROP USER $VAULT_USER;"
+
+  log "dropping the static user from oracle db"
+  docker_sqlplus_admin "DROP USER $STATIC_USER;"
+
+  log "removing the oracle docker container"
+  docker rm oracle --force
+
+  log "killing vault process"
+  pkill vault
+
+  log "END TEARDOWN"
+}
+
+@test "Read license" {
+  run vault read -format=json sys/license/status
+  assert_status 0
+}
+
+@test "POST /database/config/:name - write oracle connection config" {
+  run vault write --namespace=ns1 database/config/$DB_NAME \
+    plugin_name=vault-plugin-database-oracle \
+    connection_url="{{username}}/{{password}}@localhost:1521/ORCLPDB1" \
+    allowed_roles="*" \
+    username="$VAULT_USER" \
+    password="$VPASS" \
+    max_conneciton_lifetime="30s"
+  assert_status 0
+}
+
+@test "GET /database/config/:name - read oracle connection config" {
+  run vault read --namespace=ns1 database/config/$DB_NAME
+  assert_status 0
+}
+
+@test "LIST /database/config - list configs" {
+  run vault list --namespace=ns1 database/config
+  assert_status 0
+}
+
+@test "DELETE /database/config/:name - delete oracle connection config" {
+  run vault write --namespace=ns1 database/config/delete-me \
+    plugin_name=vault-plugin-database-oracle \
+    connection_url="{{username}}/{{password}}@localhost:1521/ORCLPDB1" \
+    allowed_roles="*" \
+    username="$VAULT_USER" \
+    password="$VPASS" \
+    max_conneciton_lifetime="30s"
+  assert_status 0
+
+  run vault delete --namespace=ns1 database/config/delete-me
+  assert_status 0
+}
+
+@test "POST /database/reset/:name - reset oracle connection" {
+  run vault write --namespace=ns1 -force database/reset/$DB_NAME
+  assert_status 0
+}
+
+@test "POST /database/roles/:name - write role" {
+  run vault write --namespace=ns1 database/roles/$VROLE \
+    db_name=$DB_NAME \
+    creation_statements='CREATE USER "{{username}}" IDENTIFIED BY "{{password}}"' \
+    creation_statements='GRANT CONNECT TO "{{username}}"' \
+    revocation_statements='DROP USER "{{username}}"' \
+    default_ttl="1h" \
+    max_ttl="1h"
+  assert_status 0
+}
+
+@test "GET /database/roles/:name - read role" {
+  run vault read --namespace=ns1 database/roles/$VROLE
+  assert_status 0
+}
+
+@test "LIST /database/roles - list roles" {
+  run vault list --namespace=ns1 database/roles
+  assert_status 0
+}
+
+@test "DELETE /database/roles/:name - delete role" {
+  run vault write --namespace=ns1 database/roles/delete-me \
+    db_name=$DB_NAME \
+    creation_statements='CREATE USER "{{username}}" IDENTIFIED BY "{{password}}"' \
+    creation_statements='GRANT CONNECT TO "{{username}}"' \
+    revocation_statements='DROP USER "{{username}}"' \
+    default_ttl="1h" \
+    max_ttl="1h"
+  assert_status 0
+
+  run vault delete --namespace=ns1 database/roles/delete-me
+  assert_status 0
+}
+
+@test "GET /database/creds/:name - generate dynamic credentials" {
+  run vault read --namespace=ns1 database/creds/$VROLE
```

Comment:
Might be nice to confirm the TTL is as expected?

Reply:
good idea!